fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/loanschedule/domain/AprCalculator.java

```diff
@@ -41,10 +41,29 @@ public BigDecimal calculateFrom(final PeriodFrequencyType interestPeriodFrequenc
             case YEARS:
                 defaultAnnualNominalInterestRate = interestRatePerPeriod.multiply(BigDecimal.valueOf(1));
             break;
+            case WHOLE_TERM:
+                        final BigDecimal ratePerPeriod = interestRatePerPeriod.divide(BigDecimal.valueOf(numberOfRepayments*repaymentEvery), 8, RoundingMode.HALF_UP);
+                         
+                         switch (repaymentPeriodFrequencyType) {
+                             case DAYS:
+                                 defaultAnnualNominalInterestRate = ratePerPeriod.multiply(BigDecimal.valueOf(365));
```

Comment:
Shoudn't this consider the no. of days in a year as per product configuration instead of 365?

Reply:
I followed what was already implemented for "DAYS" frequency from the first switch loop.